torchvision/models/inception.py

```diff
@@ -158,37 +178,46 @@ def forward(self, x):
         # N x 2048
         x = self.fc(x)
         # N x 1000 (num_classes)
-        if torch.jit.is_scripting():
-            if not aux_defined:
-                warnings.warn("Scripted InceptionNet always returns InceptionOutputs Tuple")
-            return InceptionOutputs(x, aux)
-        else:
-            return self.eager_outputs(x, aux)
+        return x, aux
 
     @torch.jit.unused
     def eager_outputs(self, x, aux):
-        # type: (torch.Tensor, Optional[torch.Tensor]) -> InceptionOutputs
+        # type: (Tensor, Optional[Tensor]) -> InceptionOutputs
         if self.training and self.aux_logits:
+            return _InceptionOutputs(x, aux)
+        else:
+            return x
+
+    def forward(self, x):
+        x = self._transform_input(x)
+        x, aux = self._forward(x)
+        aux_defined = self.training and self.aux_logits
+        if torch.jit.is_scripting():
+            if not aux_defined:
+                warnings.warn("Scripted Inception3 always returns Inception3 Tuple")
             return InceptionOutputs(x, aux)
-        return x
+        else:
+            return self.eager_outputs(x, aux)
 
 
 class InceptionA(nn.Module):
 
-    def __init__(self, in_channels, pool_features):
+    def __init__(self, in_channels, pool_features, basic_conv2d=None):
```

Comment:
Do you mean using 'block' instead of 'basic_conv2d' in all the Inception* classes, such as in this case:
```
..., block=None)
...
if block is None:
    block = BasicConv2d
...
self.branch7x7_1 = block(in_channels, c7, kernel_size=1)
self.branch7x7_2 = block(c7, c7, kernel_size=(1, 7), padding=(0, 3))
self.branch7x7_3 = block(c7, 192, kernel_size=(7, 1), padding=(3, 0))
...
```

I'm ok with this naming, just feel when users see 'block' in the class body it might be less explicit  to them than 'basic_conv2d'.